src/Utilities/ErrorHandling/CMakeLists.txt

```diff
@@ -24,19 +24,68 @@ spectre_target_headers(
   Exceptions.hpp
   ExpectsAndEnsures.hpp
   FloatingPointExceptions.hpp
+  FormatStacktrace.hpp
   StaticAssert.hpp
   )
 
 target_link_libraries(
   ${LIBRARY}
   PRIVATE
+  Boost::boost
   Charmxx::charmxx
   PUBLIC
   SystemUtilities
   )
 
-# For `backtrace` functionality
+# Expose readable function names in backtrace
+if(CMAKE_VERSION GREATER_EQUAL 3.13)
+  target_link_options(${LIBRARY} PUBLIC "-rdynamic")
+else()
+  target_link_libraries(${LIBRARY} PUBLIC -Wl,--export-dynamic)
+endif()
+
+# Try to use libbacktrace in boost::stacktrace when available. See:
+# https://www.boost.org/doc/libs/1_78_0/doc/html/stacktrace/configuration_and_build.html
+find_library(
+  BACKTRACE_LIB
+  NAMES backtrace
+  PATH_SUFFIXES lib64 lib
+  HINTS ${CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES})
+find_path(
+  BACKTRACE_HEADER_DIR
+  NAMES backtrace.h
+  PATH_SUFFIXES include
+  HINTS ${CMAKE_CXX_IMPLICIT_LINK_DIRECTORIES})
```

Comment:
This is searching for a header, so I don't think it should be looking in link directories.

Reply:
This is where the header is in the container 🤷‍♂️ I think it's distributed with gcc like this, but I couldn't find any reliable information on it. So I think all we can do is try to find it, and else print a message and fall back to Boost's standard way of parsing the backtrace (which doesn't show line numbers).